internal/urlparser.go

```diff
@@ -0,0 +1,107 @@
+package internal
+
+import (
+	"net/url"
+	"regexp"
+	"strings"
+)
+
+func ParseEngineURL(engine string) (*url.URL) {
+	if engine == "" {
+		FatalError("engine URL not specified")
+	}
+	u, err := parseEngineURL(engine)
+	if err != nil {
+		FatalErrorf("could not parse engine url '%s' got error: '%s'", engine, err)
+	}
+	return u
+}
+
+// ParseEngineURL parses the engine parameter and returns an websocket URL if at all possible
```

Comment:
The tabs messes up these comments, but the content is great!

Reply:
I saw that too! Will fix

Reply:
Only looks strange on github appearently, looks good in editores.